lib/TH/THLapack.h

```diff
@@ -3,15 +3,21 @@
 
 #include "THGeneral.h"
 
+#define THErrorHandler(X) X
+
 #define THLapack_(NAME) TH_CONCAT_4(TH,Real,Lapack_,NAME)
-#define THLapackCheck(fmt, func, info , ...)						\
-if (info < 0) {														\
-  THError("Lapack Error in %s : Illegal Argument %d", func, -info); \
-} else if(info > 0) {												\
-  THError(fmt, func, info, ##__VA_ARGS__);							\
-}																	\
 
+#define THLapackCheck(fmt, func, info , ...) \
+  THLapackCheckWithHandler(fmt, THErrorHandler(), func, info, ##__VA_ARGS__)
 
+#define THLapackCheckWithHandler(fmt, cleanup, func, info , ...)    \
```

Comment:
This should probably not be called "handler", but rather something like "cleanup" or "finally". "Handler" was just a placeholder name I came up with on short notice.